salt/fileserver/__init__.py

```diff
@@ -129,7 +129,10 @@ def check_file_list_cache(opts, form, list_cache, w_lock):
                 else:
                     # if filelist does not exists yet, mark it as expired
                     age = opts.get('fileserver_list_cache_time', 20) + 1
-                if age < opts.get('fileserver_list_cache_time', 20):
+                if age < 0:
+                    # Cache is from the future! Warn and mark cache invalid.
+                    log.warning('The file list_cache was created in the future!')
+                if age < opts.get('fileserver_list_cache_time', 20) or age < 0:
```

Comment:
This line decides to return the existing data if the condition is `True`. So it shall be:
```python
if 0 < age < opts.get('fileserver_list_cache_time', 20):
```